packages/lib-classifier/src/components/Classifier/components/TaskArea/TaskArea.js

```diff
@@ -1,100 +1,103 @@
 import { Tab, Tabs } from '@zooniverse/react-components'
 import counterpart from 'counterpart'
 import { Box } from 'grommet'
-import { inject, observer } from 'mobx-react'
-import { bool, func, object } from 'prop-types'
-import React from 'react'
+import { bool, func, object, shape, string } from 'prop-types'
+import queryString from 'query-string'
+import React, { useEffect, useRef, useState } from 'react'
 
-import Tasks from './components/Tasks'
+import { DisabledTaskPopup, Tasks } from './components'
 import en from './locales/en'
 import SlideTutorial from '../SlideTutorial'
 
 counterpart.registerTranslations('en', en)
 
 // TODO: add autofocus for the first tab/task area
-function storeMapper (stores) {
-  const { disableTutorialTab, setActiveTutorial, tutorial } = stores.classifierStore.tutorials
-
-  return {
-    disableTutorialTab,
-    setActiveTutorial,
-    tutorial
-  }
-}
+/**
+The tabbed tasks area of the classifier, with tabs for the tutorial and active tasks.
+*/
+export default function TaskArea({
+  className,
+  disableTutorialTab = true,
+  setActiveTutorial = () => true,
+  subject,
+  tutorial = null
+}) {
+  const [ activeIndex, setActiveIndex ] = useState(0)
+  const [ disabled, setDisabled ] = useState(false)
+  const taskArea = useRef(null)
 
-class TaskArea extends React.Component {
-  constructor () {
-    super()
+  useEffect(function onSubjectChange() {
+    // TODO: remove this once testing is complete.
+    const URLParams = queryString.parse(window?.location?.search)
+    const finished = (subject && URLParams?.finished) || subject?.retired || subject?.already_seen
```

Comment:
I think this will need an additional filter condition, otherwise the task area is going to be disabled for finished subjects in all projects.

Reply:
Maybe add a check for `workflow.hasIndexedSubjects`, so that this is limited to indexed subject sets for the time being.